ignite/pkg/protoanalysis/protoutil/parser.go

```diff
@@ -0,0 +1,38 @@
+package protoutil
+
+import (
+	"io"
+	"os"
+	"strings"
+
+	"github.com/emicklei/proto"
+	"github.com/emicklei/proto-contrib/pkg/protofmt"
+)
+
+// TODO: Maybe allow a cache?
```

Comment:
We should try to document every TODO in new issues and give a link to the issue in the comment
Could it also be possible to get more details on how we would use a cache for proto parser?

Reply:
I'll try and clean these up, some are relics from early brainstorming, some are refactorings or optimizations that seemed possible.

If I remember correctly, the cache idea came about because its done someplace else in the codebase. I'm unsure if it has merit though since I don't really think we load the same proto file multiple times.